apps/web/app/utils/emergency-proposals.ts

```diff
@@ -7,31 +7,30 @@ type UpgradeProposal = {
   salt: Hex;
 };
 
+export const upgradeStructAbi = {
+  type: "tuple",
+  components: [
+    {
+      type: "tuple[]",
+      components: [
+        { type: "address", name: "target" },
+        { type: "uint256", name: "value" },
+        { type: "bytes", name: "data" },
+      ],
+      name: "calls",
+    },
+    { type: "address", name: "executor" },
+    { type: "bytes32", name: "salt" },
+  ],
+} as const;
```

Comment:
Can we get this from the abis in `packages/contracts`?

Reply:
Maybe we can!

Reply:
I didn't find a clean way to extrasct an struct from the abi. Maybe we can do something else, though. I'm not sure.